### PR TITLE
r/aws_apigatewayv2_stage: Correctly handle diffs for HTTP API 'default_route_settings.logging_level' attribute, plus the special '$default' stage

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes https://github.com/terraform-providers/terraform-provider-aws/issues/12893.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/aws_apigatewayv2_stage: Correctly handle the `default_route_settings.logging_level` attribute for HTTP APIs
resource/aws_apigatewayv2_stage: Only set the `execute_arn` attribute for WebSocket APIs
resource/aws_apigatewayv2_stage: Set the correct `invoke_url` attribute for HTTP APIs and the `$default` stage name
```

Output from acceptance testing:

```
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Stage_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Stage_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Stage_basicWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Stage_basicWebSocket
=== RUN   TestAccAWSAPIGatewayV2Stage_basicHttp
=== PAUSE TestAccAWSAPIGatewayV2Stage_basicHttp
=== RUN   TestAccAWSAPIGatewayV2Stage_defaultHttpStage
=== PAUSE TestAccAWSAPIGatewayV2Stage_defaultHttpStage
=== RUN   TestAccAWSAPIGatewayV2Stage_disappears
=== PAUSE TestAccAWSAPIGatewayV2Stage_disappears
=== RUN   TestAccAWSAPIGatewayV2Stage_AccessLogSettings
=== PAUSE TestAccAWSAPIGatewayV2Stage_AccessLogSettings
=== RUN   TestAccAWSAPIGatewayV2Stage_ClientCertificateIdAndDescription
=== PAUSE TestAccAWSAPIGatewayV2Stage_ClientCertificateIdAndDescription
=== RUN   TestAccAWSAPIGatewayV2Stage_DefaultRouteSettings
=== PAUSE TestAccAWSAPIGatewayV2Stage_DefaultRouteSettings
=== RUN   TestAccAWSAPIGatewayV2Stage_Deployment
=== PAUSE TestAccAWSAPIGatewayV2Stage_Deployment
=== RUN   TestAccAWSAPIGatewayV2Stage_RouteSettings
=== PAUSE TestAccAWSAPIGatewayV2Stage_RouteSettings
=== RUN   TestAccAWSAPIGatewayV2Stage_StageVariables
=== PAUSE TestAccAWSAPIGatewayV2Stage_StageVariables
=== RUN   TestAccAWSAPIGatewayV2Stage_Tags
=== PAUSE TestAccAWSAPIGatewayV2Stage_Tags
=== CONT  TestAccAWSAPIGatewayV2Stage_basicWebSocket
=== CONT  TestAccAWSAPIGatewayV2Stage_DefaultRouteSettings
=== CONT  TestAccAWSAPIGatewayV2Stage_Tags
=== CONT  TestAccAWSAPIGatewayV2Stage_StageVariables
=== CONT  TestAccAWSAPIGatewayV2Stage_RouteSettings
=== CONT  TestAccAWSAPIGatewayV2Stage_Deployment
=== CONT  TestAccAWSAPIGatewayV2Stage_disappears
=== CONT  TestAccAWSAPIGatewayV2Stage_defaultHttpStage
=== CONT  TestAccAWSAPIGatewayV2Stage_ClientCertificateIdAndDescription
=== CONT  TestAccAWSAPIGatewayV2Stage_basicHttp
=== CONT  TestAccAWSAPIGatewayV2Stage_AccessLogSettings
--- PASS: TestAccAWSAPIGatewayV2Stage_disappears (26.40s)
--- PASS: TestAccAWSAPIGatewayV2Stage_basicWebSocket (31.48s)
--- PASS: TestAccAWSAPIGatewayV2Stage_defaultHttpStage (32.43s)
--- PASS: TestAccAWSAPIGatewayV2Stage_Tags (49.49s)
--- PASS: TestAccAWSAPIGatewayV2Stage_StageVariables (50.10s)
--- PASS: TestAccAWSAPIGatewayV2Stage_DefaultRouteSettings (50.33s)
--- PASS: TestAccAWSAPIGatewayV2Stage_RouteSettings (50.62s)
--- PASS: TestAccAWSAPIGatewayV2Stage_ClientCertificateIdAndDescription (51.49s)
--- PASS: TestAccAWSAPIGatewayV2Stage_basicHttp (57.67s)
--- PASS: TestAccAWSAPIGatewayV2Stage_Deployment (62.39s)
--- PASS: TestAccAWSAPIGatewayV2Stage_AccessLogSettings (78.87s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	78.933s
```

### CODE REVIEW

**Comment by nicksrandall** [comment]
Is this ready for review?

**Comment by aashishmodak** [comment]
Can we get this merged? This is blocking the use of stage resource

**Comment by ewbankkit** [comment]
@gdavison Did you get a chance to review this yet?

**Comment by zhaoyi0113** [comment]
Does this PR solve the issue `https://github.com/terraform-providers/terraform-provider-aws/issues/12893`?

**Comment by sc250024** [comment]
> Does this PR solve the issue `https://github.com/terraform-providers/terraform-provider-aws/issues/12893`?

@zhaoyi0113 Yes.

**Comment by zhaoyi0113** [comment]
> > Does this PR solve the issue `https://github.com/terraform-providers/terraform-provider-aws/issues/12893`?
> 
> @zhaoyi0113 Yes. Please merge 🙏

that's good to hear. I wonder when this one will be merged. Currently I can't update terraform state because of this issue. Thanks.

**Comment by luneo7** [comment]
You can bypass this error by adding:

```
lifecycle {
    ignore_changes = [deployment_id, default_route_settings]
  }
```

File: website/docs/r/apigatewayv2_stage.html.markdown [hunk 1]
```diff
@@ -76,7 +76,7 @@ In addition to all arguments above, the following attributes are exported:
 or in an [`aws_iam_policy`](/docs/providers/aws/r/iam_policy.html) to authorize access to the [`@connections` API](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-how-to-call-websocket-api-connections.html).
 See the [Amazon API Gateway Developer Guide](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-control-access-iam.html) for details.
 * `invoke_url` - The URL to invoke the API pointing to the stage,
-  e.g. `wss://z4675bid1j.execute-api.eu-west-2.amazonaws.com/example-stage`
+  e.g. `wss://z4675bid1j.execute-api.eu-west-2.amazonaws.com/example-stage`, or `https://z4675bid1j.execute-api.eu-west-2.amazonaws.com/`
```
Comment (anGie44) [inline website/docs/r/apigatewayv2_stage.html.markdown:79]: I would also just note in the `execution_arn` bullet point item that the value is only set/computed for WebSocket APIs (as it was previously populated for both types)

Comment (ewbankkit) [reply website/docs/r/apigatewayv2_stage.html.markdown:80]: Done.

**Comment by reyescameron17** [comment]
I upgraded to aws provider "2.65.0" but I am still running into this issue when trying to use `aws_apigatewayv2_stage` resource. Is there also a configuration I need to make in the `route_settings` block to resolve this?

**Pull request merged**